src/System.Private.CoreLib/shared/System/Text/Utf16Utility.cs

```diff
@@ -159,5 +168,39 @@ internal static bool UInt32OrdinalIgnoreCaseAscii(uint valueA, uint valueB)
 
             return (((combinedIndicator >> 2) | ~0x00200020u) & differentBits) == 0;
         }
+
+        /// <summary>
+        /// Given two UInt64s that represent four ASCII UTF-16 characters each, returns true iff
+        /// the two inputs are equal using an ordinal case-insensitive comparison.
+        /// </summary>
+        /// <remarks>
+        /// This is a branchless implementation.
+        /// </remarks>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        internal static bool UInt64OrdinalIgnoreCaseAscii(ulong valueA, ulong valueB)
+        {
+            // ASSUMPTION: Caller has validated that input values are ASCII.
+            Debug.Assert(AllCharsInUInt64AreAscii(valueA));
+            Debug.Assert(AllCharsInUInt64AreAscii(valueB));
+
+            // a mask of all bits which are different between A and B
+            ulong differentBits = valueA ^ valueB;
+
+            // the 0x80 bit of each word of 'lowerIndicator' will be set iff the word has value < 'A'
+            ulong lowerIndicator = valueA + 0x0100010001000100ul - 0x0041004100410041ul;
```

Comment:
Nit, up to you, but underscores are nice when we're at 16 digits ... 0x0100_0100_0100_0100ul